crates/ecs/src/lib.rs

```diff
@@ -265,74 +259,293 @@ impl<'systems> Schedule<'systems> for Unordered<'systems> {
     }
 }
 
+/// Stores components associated with entity ids.
+#[derive(Debug, Default)]
+struct Archetype {
+    component_typeid_to_component_vec: HashMap<TypeId, Box<dyn ComponentVec>>,
+    entity_id_to_component_index: HashMap<usize, usize>,
+    last_entity_id_added: usize,
+}
+
+impl Archetype {
+    /// Adds an `entity_id` to keep track of and store components for.
+    fn store_entity(&mut self, entity_id: usize) {
+        let entity_index = self.entity_id_to_component_index.len();
+
+        self.component_typeid_to_component_vec
+            .values_mut()
+            .for_each(|v| v.push_none());
```

Comment:
It is a temporary solution. But I'm not sure how it can be solved. Imagine that you store an entity in an archetype, but you have not yet transferred all components to the archetypes respective component vectors. All indices need to line up for each entity, but if you only add some of the components the indices will quickly become uneven. If we are to remove Option, we need some way of guaranteeing that the indices do not become unaligned.

Reply:
>Imagine that you store an entity in an archetype, but you have not yet transferred all components to the archetypes respective component vectors.

What do you mean "have not yet transferred all components"? Do you mean that if I add an entity with components `A`, `B` and `C` there will be some nanoseconds between when I add `A`, `B` and `C` to their respective component vecs? That shouldn't be an issue since mutating an archetype like that should require exclusive access to it, right?

Reply:
Nevermind, its to do with `BigArchetype`, but it needs to exist right now, but should be removed in #72. I'll add a commet